### DISCUSSION TITLE
Issue to get width and height using `vips_pdfload`

### DISCUSSION THREAD

**Discussion by betorcs** [libvips/libvips#3938] [Q&A]
I have a JNA integration with vips library. It has been worked quite well. 

Recently I'm having a problem with few PDF files on load them thrus `vips_loadpdf` with parameters `scale` and `page` is returning `0` in attributes `width` and `height`,  others operations like `vips_dzsave` works fine.

I'd like to know if there some information or a guide how can I debug it

Unfornutally I'm not allowed share the files

**Comment by jcupitt** [comment]
Hello @betorcs,

libvips usually uses poppler to load PDFs, the same library that evince uses for PDF preview, so I'd try viewing your PDFs in that and see if they work.

You could try updating libvips and libpoppler, there have been some improvements to PDF handling.

You could also try an alternative PDF library. libvips can also use PDFium, the same library that Chrome uses.

**Reply by betorcs** [reply]
@jcupitt thank you for reply.

- It's using PDFium 
- I already upgraded from `8.12` to `8.15`, no success

The strange think I forgot to mention the issue happen in AWS Lambda in arch `x86`, but running the same code in my local `MacOS M2 arm64` I just see the issue happening once for a single file.

**Reply by jcupitt** [reply]
Huh that IS odd. I guess these files view ok in Chrome? Which PDFium build are you using?

You could try poppler as well. You probably can't use it in production due to the licence, but it might help narrow the issue down.
